### PR TITLE
Add foreman_pipeline to jjb

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman_pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- project:
+    name: foreman_pipeline
+    repo: foreman_pipeline
```
Comment (domcleal) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman_pipeline.yaml]: This is the default (taken from the name)

**Comment by domcleal** [comment]
Are you following up with PR testing after?

**Comment by xprazak2** [comment]
@domcleal: I would very much like that

**Comment by domcleal** [comment]
Sure, you can either do it now via the second set of instructions on http://projects.theforeman.org/projects/foreman/wiki/Jenkins#Adding-a-new-Foreman-plugin if your tests are passing OK, or we can merge this to see if it works (once that comment's fixed) and enable PR tests after.

**Comment by xprazak2** [comment]
Everything I need to test PRs shlould be configured now, except for the  `value of GITHUB_SECRET_TOKEN from rhc env list -a prprocessor -n theforeman` in repo webhook, may I be so bold and ask for help, @domcleal ?

File: puppet/modules/slave/templates/test_pull_requests_foreman_pipeline.json.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "proxy_port": null,
+    "jenkins_host": "http://ci.theforeman.org/",
+    "jenkins_url": "http://ci.theforeman.org/",
+    "hub_config": "~/.config/hub",
+    "github_user": "theforeman",
+    "bot_github_user": "<%= github_user -%>",
+    "irc_channel": "#theforeman-dev",
+    "not_mergable_cache": "~/pr_tests/cache/test_pull_request_foreman_pipeline_not_mergable",
+    "repo_groups": {
+        "foreman_pipeline": "foreman_pipeline",
```
Comment (domcleal) [inline puppet/modules/slave/templates/test_pull_requests_foreman_pipeline.json.erb]: The trailing comma here isn't valid JSON.

**Comment by domcleal** [comment]
Sure, I'll add the webhook on merge, don't worry about that.

**Comment by domcleal** [comment]
Oh sorry, I didn't know you're updated this.  In light of https://github.com/theforeman/foreman-packaging/pull/892 I'm unsure if it'll work as-is because our jobs only ever install Foreman and add the plugin in on top.  If you can specify Katello as a dependency (Gemfile, gemspec) and a regular db:migrate/seed works then it might be OK.

**Comment by xprazak2** [comment]
@domcleal: I updated the foreman_pipeline gemspec with Katello dependency, if there is anything else, let me know...

**Comment by domcleal** [comment]
Thanks, it should show up in an hour or two under the Plugins tab on Jenkins.

**Pull request merged**